banzaicloud/logging-operator
# SQS output plugin

| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | none
| License         | Apache 2.0


### What's in this PR?
Adds [SQS output plugin](https://github.com/ixixi/fluent-plugin-sqs)

### Additional context
To test the implementation:
- compiled `fluentd` with the added `fluent-plugin-sqs` gem and pushed up the docker image to my public repository
- Modified `pkg/sdk/api/v1beta1/logging_types.go` and set `DefaultFluentdImageRepository` to my public image repo temporarily
- compile the controller and pushed up to my public repository
- deployed to my local kind cluster
- followed the [guide](https://banzaicloud.com/docs/one-eye/logging-operator/quickstarts/es-nginx/) and deployed the logging demo
- added the following output
```
apiVersion: logging.banzaicloud.io/v1beta1
kind: Output
metadata:
  name: sqs-output-sample
  namespace: logging
spec:
  sqs:
    queue_name: some-aws-sqs-queue
    create_queue: false
    region: us-east-1
```
- modified the existing `Flow` and appended `sqs-output-sample`

I was able to verify that messages were successfully being populated in `some-aws-sqs-queue` queue

### Checklist


- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)
- [x] User guide and development docs updated (if needed)
- [x] Related Helm chart(s) updated (if needed)
